docs/P4c-test-documentation.md

```diff
@@ -0,0 +1,235 @@
+# **Adding New Tests to the P4C Repository**
+
+**This document provides a guide for developers who want to add new tests to the P4C repository. It explains where to place test files, how to manage expected output files, and how to handle tests that are expected to fail (Xfail).**
+
+---
+
+
+
+
+
+## **1. Adding "Positive" Tests**
+
+Positive tests are those where the P4 compiler should process the input without any errors.
+
+### Steps for Adding Positive Tests
+
+1. **Source File Location**:
+
+   - Positive tests are located in multiple directories based on the purpose and the backend/tool being tested. Below is a description of the primary directories under `p4c/testdata` and their purposes:
+
+     - **`p4_16_samples`**: Contains general positive tests for the P4\_16 language.
+     - **`p4_16_samples_outputs`**: Stores the expected outputs for the tests in `p4_16_samples`.
+     - **`p4_14_samples`**: Contains tests specific to the older P4\_14 language version.
+     - **`p4_14_samples_outputs`**: Stores the expected outputs for tests in `p4_14_samples`.
+     - **`bmv2`**: Holds tests targeting the BMv2 backend, including packet-processing logic and examples.
```

Comment:
Before this list of items, it implies this is a list of "multiple directories".  I know where to find the directories `p4_14_samples`, `p4_16_samples`, etc. listed above, but where are the directories `bmv2` and `psa` you are referring to?

Reply:
Also `xdp`, `arch`, `error_samples`, `stf_samples`, `stress_tests`?